lib/use-checkout-pricing.js

```diff
@@ -0,0 +1,100 @@
+import { useEffect, useState } from 'react';
+import useRecurly from './use-recurly';
+import cloneDeep from 'lodash/cloneDeep';
+
+/**
+ * useCheckoutPricing interface
+ *
+ * Used to describe:
+ *  1. `initialInputs` argument in `useCheckoutPricing`
+ *  2. `newInputs` argument in `updateInputs`
+ *  3. `useEffect` dependencies
+ *
+ * @typedef {Object} useCheckoutPricingInterface
+ * @property {Array} subscriptions
+ */
+
+/**
+ * A custom hook for interfacing with react-recurly's checkoutPricing API meant to mimic the call signature, return
+ * type, and behavior of `react.useState`.
+ *
+ * Accepts an `initialInputs` param (same as useState) and an error handling function.
+ *
+ * Returns a tuple with an output object and an update function similar to useState.
+ *
+ * @typedef {Object} output
+ * @property {Object} price
+ * @property {Object} pricing
+ * @property {Boolean} loading
+ *
+ * @typedef {Function} setPricing
+ * @typedef {[output, setPricing]} useCheckoutPricingInstance
+ *
+ * @param {PricingInput} useCheckoutPricingInterface
+ * @param {function} handleError
+ * @returns {useCheckoutPricingInstance} useCheckoutPricingInstance
+ */
+export default function useCheckoutPricing(initialInputs = {}, handleError = throwError) {
```

Comment:
After some deliberation, I think this is the most logical and conventional architecture for the params.